src/test/java/com/flightstats/hub/util/IntegrationUdpServer.java

```diff
@@ -1,74 +1,71 @@
 package com.flightstats.hub.util;
 
 import lombok.Builder;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import lombok.extern.slf4j.Slf4j;
 
 import java.io.BufferedReader;
 import java.io.ByteArrayInputStream;
 import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.net.DatagramPacket;
 import java.net.DatagramSocket;
 import java.io.IOException;
-import java.util.concurrent.TimeUnit;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
 import java.util.stream.Collectors;
 
+@Slf4j
 @Builder
 public class IntegrationUdpServer {
-    private final static Logger logger = LoggerFactory.getLogger(IntegrationUdpServer.class);
-    private long timeoutMillis;
-    private final List<String> queue = new ArrayList<>();
     private int port;
     private boolean listening;
+    private CountDownLatch startupCountDownLatch;
+    private ExecutorService executorService;
 
     private final Map<String, String> store = new HashMap<>();
 
-    private final CompletableFuture<Map<String, String>> serverFuture = CompletableFuture.supplyAsync(() -> {
-        try {
-            DatagramSocket serverSocket = new DatagramSocket(port);
-            while (listening) {
-                byte[] data = new byte[70];
-                String result = openSocket(data, serverSocket);
-                Thread.sleep(500);
-                addValueToStore(result);
-            }
-        } catch(Exception ex) {
-            listening = false;
-        }
-//        logger.info(":::::::::, {}", store);
-        return store;
-    });
+    public CompletableFuture<Map<String, String>> getServerFuture() {
+        return  CompletableFuture.supplyAsync(() -> {
+            try {
+                DatagramSocket serverSocket = new DatagramSocket(port);
+                while (listening) {
+                    byte[] data = new byte[70];
```

Comment:
If you know what the magic number 70 means, it's worth extracting a constant while you're here.

Reply:
I'm afraid I don't know. @Paul-Hess?

Reply:
At some point it was the number of bytes in the fairly static string sent to the server